### DISCUSSION TITLE
Searching by Person, Persons, Subject or Subjects API - Not Yielding Correct Results

### DISCUSSION THREAD

**Discussion by vladisimo** [photoprism/photoprism#4647] [General]
When I search manually using Person, Persons, Subject or Subjects via the web interface for 3, 4, 5 or more people's names, the search yields a number of photos. When I do a search via the API, it appears that beyond searching for 2 names, the search either returns a single result, or results which are not accurate (i.e. the photo should contain say 3 people with the names defined, but only two people are named). Is there a limitation on the API whereby it only searches a maximum of 2 people in any photo? If so, how can I bypass this so I can search for more than 2 people? I need to have this working as I am using another app with face detection to provide a slideshow of visitors in my house.

**Comment by omerdduran** [comment]
Hi. There should be no inherent limitation in the API that limits searches to 2 people. The API uses the same search backend as the web interface.

Based on your description, I suspect it's related to how you're constructing the search query. When searching for multiple people using the API, it's important to use the correct syntax for combining names.

You can use either the people filter or the person filter:
- people - more flexible matching
- person- Exact name matches

https://docs.photoprism.app/user-guide/search/filters/#using-search-filters
